core/trino-main/src/test/java/io/trino/util/JoinParamUtil.java

```diff
@@ -0,0 +1,139 @@
+package io.trino.util;
+
+import io.trino.sql.planner.assertions.ExpectedValueProvider;
+import io.trino.sql.planner.assertions.PlanMatchPattern;
+import io.trino.sql.planner.plan.JoinNode;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
+
+public class JoinParamUtil
```

Comment:
instead of having `JoinParamUtil` and `JoinParamBuilder` you could have a single `JoinMatcherBuilder` that either builds `JoinMatcher` or `PlanMatchPattern`